enso/experiment/__init__.py

```diff
@@ -162,8 +170,33 @@ def _run_sub_experiment(
 
     def _run_experiment(self, dataset_name, current_setting, experiments):
         """Responsible for running all experiments specified in config."""
-        results = pd.DataFrame(columns=self.columns)
         dataset = self._load_dataset(dataset_name, current_setting.get("Featurizer"))
+        if EXPERIMENT_PARAMS:
+            exp_params = deepcopy(EXPERIMENT_PARAMS)
+            # add any experiments that were not included
+            for experiment_cls in experiments:
+                if experiment_cls.__name__ not in exp_params:
+                    exp_params[experiment_cls.__name__] = {}
+            # add the 'All' configuration to all the experiments
+            exp_params = {('All' if k.upper() == 'ALL' else k):v for k, v in exp_params.items()}
+            if 'All' in EXPERIMENT_PARAMS.keys():
```

Comment:
I would think you would want this to behave in the opposite fashion -- i.e. specific experiment hyperparam definitions take precedence over the "all" key.  

So something more like:

```
exp_params[k] = copy.deepcopy(EXPERIMENT_PARAMS['All'])
exp_params[k].update(v)
```

This way you could specify a more narrow range of hyperparams for a specific experiment if you know the full range is probably overkill or something.

Reply:
:+1: